src/genericfit.jl

```diff
@@ -30,5 +30,15 @@ fit_mle(dt::Type{D}, x::AbstractArray, w::AbstractArray) where {D<:UnivariateDis
 fit_mle(dt::Type{D}, x::AbstractMatrix) where {D<:MultivariateDistribution} = fit_mle(D, suffstats(D, x))
 fit_mle(dt::Type{D}, x::AbstractMatrix, w::AbstractArray) where {D<:MultivariateDistribution} = fit_mle(D, suffstats(D, x, w))
 
+"""
+    fit(D, x)
+
+Fit a distribution of type `D` to a dataset `x` using an arbitrary algorithm.
+
+The fit function will choose a reasonable way to fit the distribution, which,
+in most cases, is maximum likelihood estimation. Note that this algorithm may
```

Comment:
Maybe this can be formulated in a stronger way - the default fallback is `fit_mle`, so `fit` is always the MLE fit if not explicitly defined otherwise.